src/software/ai/hl/stp/tactic/crease_defender/crease_defender_fsm.h

```diff
@@ -70,14 +90,36 @@ struct CreaseDefenderFSM
         DEFINE_SML_STATE(MoveFSM)
         DEFINE_SML_EVENT(Update)
         DEFINE_SML_SUB_FSM_UPDATE_ACTION(blockThreat, MoveFSM)
+        DEFINE_SML_STATE(DribbleFSM)
+        DEFINE_SML_GUARD(ballNearbyWithoutThreat)
+        DEFINE_SML_SUB_FSM_UPDATE_ACTION(prepareGetPossession, DribbleFSM)
 
         return make_transition_table(
             // src_state + event [guard] / action = dest_state
-            *MoveFSM_S + Update_E / blockThreat_A, MoveFSM_S = X,
+            *MoveFSM_S + Update_E[ballNearbyWithoutThreat_G] / prepareGetPossession_A =
+                DribbleFSM_S,
+            MoveFSM_S + Update_E / blockThreat_A, MoveFSM_S = X,
+            DribbleFSM_S + Update_E[!ballNearbyWithoutThreat_G] / blockThreat_A =
+                MoveFSM_S,
+            X + Update_E[ballNearbyWithoutThreat_G] / prepareGetPossession_A =
+                DribbleFSM_S,
             X + Update_E / blockThreat_A = MoveFSM_S);
     }
 
    private:
+    /** Max ratio between distances (crease and ball) / (crease and nearest enemy) for
+     * crease to chase ball. Scale from (0, 1)
+     * |------------------------------------------|
+     * | Crease | <-----------------------> Enemy |
+     * |        | <----> Ball                 |   |
+     * |       ()         x                  ()   |
+     * |------------------------------------------|
+     */
```

Comment:
Love the addition of the diagram! If you want to make it even prettier, you could look at the ASCII characters used in this diagram: https://github.com/UBC-Thunderbots/Software/blob/99e9e9e2a883d01122a377dbee8a12b7d50d659f/src/software/thunderscope/proto_unix_io.py#L23-L29